source/includes/authentication.cpp

```diff
@@ -0,0 +1,72 @@
+// start-scram-sha-256
+#include <mongocxx/client.hpp>
+#include <mongocxx/uri.hpp>
+
+auto uri = mongocxx::uri("mongodb://<username>:<password>@<hostname>:<port>/?"
+                         "authSource=admin&authMechanism=SCRAM-SHA-256");
+mongocxx::client client(uri);
+// end-scram-sha-256
+
+// start-scram-sha-1
+#include <mongocxx/client.hpp>
+#include <mongocxx/uri.hpp>
+
+auto uri = mongocxx::uri("mongodb://<username>:<password>@<hostname>:<port>/?"
+                         "authSource=admin&authMechanism=SCRAM-SHA-1");
+mongocxx::client client(uri);
+// end-scram-sha-1
+
+// start-x509
+#include <mongocxx/client.hpp>
+#include <mongocxx/uri.hpp>
+
+auto uri = mongocxx::uri("mongodb://<hostname>:<port>/?"
+                         "tls=true&tlsCertificateKeyFile=path/to/client.pem&authMechanism=MONGODB-X509");
+mongocxx::client client(uri);
+// end-x509
+
+// start-aws-connection-uri
+#include <mongocxx/client.hpp>
+#include <mongocxx/uri.hpp>
+
+auto uri = mongocxx::uri("mongodb://<AWS IAM access key ID>:<AWS IAM secret access key>@<hostname>:<port>/?"
+                         "&authMechanism=MONGODB-AWS");
```

Comment:
```suggestion
auto uri = mongocxx::uri("mongodb://<AWS IAM access key ID>:<AWS IAM secret access key>@<hostname>:<port>/?"
                         "authMechanism=MONGODB-AWS");
```

Remove extra `&` to prevent error: `URI option "" contains no "=" sign: an invalid MongoDB URI was provided`

Reply:
Good catch, thanks!